src/upload/system/library/transbank/utils/LogHandler.php

```diff
@@ -0,0 +1,139 @@
+        $this->logger->pushHandler(new RotatingFileHandler("{$this->logDir}/log_transbank_{$ecommerce}.log", 10, Logger::DEBUG));
+    }
+
+    /**
+     * Formats the size of a file in readable format.
+     * @param string $path
+     * @return string
+     */
+    private function formatBytes($path): string
+    {
+        $bytes = @filesize($path);
+        if ($bytes > 0) {
+            $unit = intval(log($bytes, 1024));
+            $units = array('B', 'KB', 'MB', 'GB');
+            if (array_key_exists($unit, $units) === true) {
+                return sprintf('%d %s', $bytes / pow(1024, $unit), $units[$unit]);
+            }
+        }
+        return '0 B';
+    }
+
+    /**
+     * Returns the directory where the logs are stored.
+     * @return string
+     */
+    private function getLogDir(): string
+    {
+        return $this->logDir;
+    }
+
+    /**
+     * Generates a list of log files.
+     * @return array|null
+     */
+    private function setLogList(): array
+    {
+        $arr = array_diff(scandir($this->logDir), array('.', '..'));
+        foreach ($arr as $value) {
+            chmod($this->logDir . "/" . $value, 0660);
+            $logList[] = "<a href='{$this->logURL}/{$value}' download>{$value}</a>";
+        }
+        return $logList ?? null;
+    }
+
+    /**
+     * Gets the information of the last log file.
+     * @return array
+     */
+    private function getLastLog(): array
+    {
+        $files = glob($this->logDir . "/*.log");
+        if (!$files) {
+            return ["No existen Logs disponibles"];
+        }
+        $files = array_combine($files, array_map("filemtime", $files));
+        arsort($files);
+        $this->lastLog = key($files);
+        $logContent = @file_get_contents($this->lastLog) ?: null;
+        return [
+            'log_file' => basename($this->lastLog),
+            'log_weight' => $this->formatBytes($this->lastLog),
+            'log_regs_lines' => count(file($this->lastLog)),
+            'log_content' => $logContent
+        ];
+    }
+
+    /**
+     * Count the number of logs in the directory.
+     * @return array
+     */
+    private function setLogCount(): array
+    {
+        $logList = $this->setLogList();
+        $count = isset($logList) ? count($logList) : 0;
+        return ['log_count' => $count];
+    }
+
+    /**
+     * Gets a summary of the current configuration and logs.
+     * @return string
+     */
+    public function getResume(): bool|string
+    {
+        $result = array(
+            'log_dir' => $this->getLogDir(),
+            'logs_count' => $this->setLogCount(),
+            'logs_list' => $this->setLogList(),
+            'last_log' => $this->getLastLog(),
+        );
+        return json_encode($result);
+    }
+
+    /**
+     * Print DEBUG log
+     */
+    public function logDebug($msg)
+    {
+
+            $this->logger->debug($msg);
+
+    }
+
+     /**
+     * Print INFO log
+     */
```

Comment:
Use the same documentation format as other methods.

Reply:
Documentation update will be performed